salt/utils/process.py

```diff
@@ -1084,6 +1097,107 @@ def _handle_signals(self, signum, sigframe):
                                 self.pid,
                                 os.getpid(),
                             )
+
```

Comment:
The coupling of gitfs logic to the process class seems a bit weird to me. I.E. having anything with a 'gitfs' name in `salt.utils.process` makes me cringe a bit. I'd prefer we find a way to keep `salt.utils.process` as generic as possible. Maybe having a way to register a signal handler? I thought this was supported already.

@s0undt3ch thoughts?

Reply:
It is not so much coupling but cleanup, checking for zombie processes, and in particular any gitfs zombie processes

Reply:
I am looking to write a clairvoyant process that knows ahead of time that it will be using GitFS and hence register a cleanup process for it  :roll_eyes: 
Seriously, thinking to check if process has a registered cleanup when some process first creates a GitFS object, and if not, register it, actually could just go ahead and register anyway since it notes if one already there and ignores registrations.
Do either of you @dwoz , @s0undt3ch see anything wrong with this approach ?

Reply:
@dwoz @s0undt3ch So this method https://github.com/saltstack/salt/blob/6052a1ede23bfe2d8dfed8fe6d53ded8796bbb0d/salt/utils/process.py#L1041
Doesn't do what you think it does, this is not getting called when a process terminates, but  more about forking etc.  Added additional debugging code to confirm this, since current debugging code is not showing it getting called.

The code in signal handling does something different, and I am up for doing something more generic, perhaps adding a call to _finalize_methods in the Signal handling, since the Signal Handler implies class Process and the method should exist.

Interested in thoughts here, since understand directly going through GitFS files is a bit of a hack, but it works :roll_eyes:

Reply:
Refactored this and no longer have explicit GitFS handling with SIGTERM, but a generalized method